frontend/src/pages/projects/projectSharing/useGroups.ts

```diff
@@ -3,17 +3,19 @@ import useFetchState, { FetchState } from '~/utilities/useFetchState';
 import { listGroups } from '~/api';
 import { GroupKind } from '~/k8sTypes';
 
+const EMPTY: GroupKind[] = [];
```

Comment:
And the tests will catch things that we don't think about!

Was this for purity or did it actually fail? Because you may be hiding a `useCallback` recompute error if the ref integrity really did change.

Reply:
After talking to @christianvogt I believe this needs to be undone unless he can provide value beyond saving ref stability issues -- since it should always allow for that to change.

Reply:
I think you're right. A call to `refresh` would expect a new value regardless if it's the same.